apiclient/rancher_api/api.py

```diff
@@ -56,8 +56,16 @@ def cluster_version(self):
         if not self._version:
             code, data = self.settings.get('server-version')
             ver = data['value']
-            # XXX: fix master-xxx-head to 8.8.8, need the API fix the problem
-            self._version = parse_version('8.8.8' if 'master' in ver else ver)
+
+            try:
+                # XXX: fix master-xxx-head to 8.8.8, need the API fix the problem
+                cur, _, _ = ver.split('-')
+                cur = "v8.8.8" if "master" in cur else cur
+                self._version = parse_version(f"{cur}.99")
```

Comment:
Is this for branch head?
e.g. `v1.4-ba586eaf-head` -> `v1.4.99`
```
>>> ver = "v1.4.0"
>>> parse_version('8.8.8' if 'master' in ver else ver)
<Version('1.4.0')>    # OK
>>> 
>>> ver = "v1.4.0-rc1"
>>> parse_version('8.8.8' if 'master' in ver else ver)
<Version('1.4.0rc1')>    # OK
>>> 
>>> ver = "v1.4.0-dev-20240101"
>>> parse_version('8.8.8' if 'master' in ver else ver)
<Version('1.4.0.dev20240101')>    # OK
>>> 
>>> ver = "master"
>>> parse_version('8.8.8' if 'master' in ver else ver)
<Version('8.8.8')>    # OK
>>> 
>>> ver = "v1.4-ba586eaf-head"
>>> parse_version('8.8.8' if 'master' in ver else ver)
Traceback (most recent call last):
  ...
    raise InvalidVersion(f"Invalid version: '{version}'")
pkg_resources._vendor.packaging.version.InvalidVersion: Invalid version: 'v1.4-ba586eaf-head'
>>> 
>>> cur, _, _ = ver.split('-')
>>> cur = "v8.8.8" if "master" in cur else cur
>>> parse_version(f"{cur}.99")
<Version('1.4.99')>
```

Reply:
I copied this code from the harvester API part assuming that Rancher follows a similar release naming. I updated it to match their release naming as best I could.